src/client/components/pages/entities/wikipedia-extract.tsx

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (C) 2023  David Kellner
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+import {Col, Row} from 'react-bootstrap';
+import {WikipediaArticleExtract, buildWikipediaUrl, getWikidataId} from '../../../../common/helpers/wikimedia';
+import DOMPurify from 'isomorphic-dompurify';
+import type {EntityT} from 'bookbrainz-data/lib/types/entity';
+import React from 'react';
+import {getAliasLanguageCodes} from '../../../../common/helpers/utils';
+import {uniq} from 'lodash';
+
+
+type State = Partial<WikipediaArticleExtract>;
+
+type Props = {
+	entity: EntityT,
+	articleExtract?: WikipediaArticleExtract,
+};
+
+
+/**
+ * Fetches a Wikipedia extract for the given Wikidata item.
+ * @param {string} wikidataId - Wikidata item ID.
+ * @param {string[]} preferredLanguages - List of language codes, preference in descending order.
+ */
+async function getWikipediaExtractForWikidata(wikidataId: string, preferredLanguages = ['en']) {
+	const apiUrl = new URL(`/wikidata/${wikidataId}/wikipedia-extract`, document.location.href);
+	apiUrl.search = new URLSearchParams(preferredLanguages.map((lang) => ['language', lang])).toString();
+	const response = await fetch(apiUrl);
+	return response.json() as Promise<WikipediaArticleExtract>;
+}
+
+
+class WikipediaExtract extends React.Component<Props, State> {
+	constructor(props: Props) {
+		super(props);
+		this.state = props.articleExtract ?? {};
+	}
+
+	componentDidMount() {
+		if (!this.state.extract) {
+			const wikidataId = getWikidataId(this.props.entity);
+			if (!wikidataId) {
+				return;
+			}
+
+			const aliasLanguages = getAliasLanguageCodes(this.props.entity);
+			const preferredLanguages = uniq(['en', ...aliasLanguages]);
+
+			getWikipediaExtractForWikidata(wikidataId, preferredLanguages).then((result) => {
```

Comment:
This should also have a `.catch(…` clause, in case the fetch request or the subsequent json parsing fail

Reply:
Good catch, I hadn't really thought about this scenario because an exception doesn't really cause any harm here (because it happens only inside a browser when we're already done).

Did you have any specific error handling in mind?
I'm asking because I had to disable ESLint in order to be able to log the error to `console` in 73e0201 (the `log` module is only for Node.js) and I don't think we want to do more for a non-required component that fails to load (and that should only happen rarely).

Reply:
Yes, console.error or console.log is fine (that is, as an exception rather than the rule).
We don't currently have a better reporting system in place.